plugin/storage/integration/token_propagation_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build token_propagation
+
+package integration
+
+import (
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const bearerToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhZG1pbiIsIm5hbWUiOiJKb2huIERvZSIsImlhdCI"
+const bearerHeader = "Bearer " + bearerToken
+
+func MockingESEToken(t *testing.T, stop *sync.WaitGroup)  {
+	http.HandleFunc("/", func (w http.ResponseWriter, r *http.Request) {
+		authValue := r.Header.Get("Authorization")
+		shouldReceive := "Bearer " + bearerToken
+		assert.Equal(t, shouldReceive, authValue)
+		stop.Done()
+	})
+	http.ListenAndServe(":9200", nil)
+}
+
+func Test_bearTokenPropagationHandler(t *testing.T)  {
+	testCases := []struct {
+		name        string
+		headerValue string
+		headerName  string
+	}{
+		{ name:"Bearer token", headerName:"Authorization", headerValue: bearerHeader},
+		{ name:"Raw Bearer token", headerName:"Authorization", headerValue: bearerToken},
+		{ name:"X-Forwarded-Access-Token", headerName:"X-Forwarded-Access-Token", headerValue: bearerHeader},
+	}
+	stop := &sync.WaitGroup{}
+
+	go MockingESEToken(t, stop)
+	for _, testCase := range testCases {
+		/* Mocking ES */
+		queryHost := "127.0.0.1:16686"
+		stop.Add(1)
+		req, err := http.NewRequest("GET", "http://"+ queryHost + "/api/services", nil)
+		req.Header.Add(testCase.headerName, testCase.headerValue)
+		assert.NoError(t, err)
+		client := &http.Client{}
+		_, err = client.Do(req)
```

Comment:
I don't think this is a valid test, the query service can simply ignore the tokens, the test is not validating anything.

Reply:
What do you mean? If query service ignore the token, the handler in the elasticsearch mocker won't received it and the test will fail.

Reply:
but the ESMock doesn't return any sensible results, the query service will fail anyway

Reply:
Ha sure, I'm not validating if the query service returns results or not,  I only validate that the token is being propagated to the ES storage. I of course can implement a better mock of ES if you think we need to validate  the query reply

Reply:
Due to the nature of the e2e tests I decided to follow your recommendation, now I returned an empty result if with an http 200 if the token was received with mocked ES, and 403 (forbidden http) otherwise.

So the test validates that the query service propagated the token correctly.

Reply:
Is the test better now?